test/jdk/java/awt/Scrollbar/AquaLFScrollbarTest/ScrollBarBorderTest.java

```diff
@@ -0,0 +1,132 @@
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import javax.swing.JScrollBar;
+import javax.swing.JPanel;
+import javax.swing.JFrame;
+import javax.swing.BoxLayout;
+import javax.swing.JLabel;
+import javax.swing.Box;
+import javax.swing.UIManager;
+import javax.swing.border.Border;
+import javax.swing.border.EmptyBorder;
+import java.awt.*;
+import java.awt.event.MouseAdapter;
+import java.awt.event.MouseEvent;
+
+/*
+ * @test
+ * @bug 8190264
+ * @summary JScrollBar ignores its border when using macOS Mac OS X Aqua look and feel
+ * @run main ScrollBarBorderTest
+ */
+public class ScrollBarBorderTest {
+
+    // On macOS 10.12.6 using the Mac look and feel (com.apple.laf.AquaLookAndFeel)
+    // the scroll bar ignores the custom border and allows the scroll thumb to move
+    // beneath the border. Run with:
+    // java ScrollBarBorderTest
+
+    // If run using any other look and feel (e.g. Metal) then the right side of
+    // the scroll bar stops at the border as expected. Run with:
+    // java -Dswing.defaultlaf=javax.swing.plaf.metal.MetalLookAndFeel ScrollBarBorderTest
+
+    // Java version: 1.8.0_151
+
+    private static JScrollBar scrollBar;
+    private static JPanel panel;
+    private static JFrame frame;
+    private static Robot robot;
+
+    public void createAndShowGUI() {
+        // create scroll bar
+        scrollBar = new JScrollBar(Scrollbar.HORIZONTAL);
+        scrollBar.setBorder(new CustomBorder());
+
+        // create panel
+        panel = new JPanel();
+        panel.setLayout(new BoxLayout(panel, BoxLayout.Y_AXIS));
+        panel.setBorder(new EmptyBorder(20, 20, 20, 20));
+        panel.add(new JLabel(UIManager.getLookAndFeel().toString()));
+        panel.add(Box.createVerticalStrut(20));
+        panel.add(scrollBar);
+
+        // create frame
+        frame = new JFrame("ScrollBarBorderTest");
+        frame.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
+        frame.getContentPane().add(panel);
+        frame.pack();
+        frame.setVisible(true);
+    }
+
+
+    private static void cleanUp() {
+        frame.dispose();
+    }
+
+    public static void main(String[] args) {
+        ScrollBarBorderTest borderTest = new ScrollBarBorderTest();
+        borderTest.createAndShowGUI();
```

Comment:
You're using Swing which is not thread-safe. You should create and access Swing components from the Event Dispatch Thread only.